submarine-workbench/workbench-server/src/test/java/org/apache/submarine/server/AbstractWorkbenchServerTest.java

```diff
@@ -0,0 +1,304 @@
+          confDir.getAbsolutePath());
+      notebookDir = new File(workbenchServerHome.getAbsolutePath() + "/notebook_" + testClassName);
+      if (cleanData) {
+        FileUtils.deleteDirectory(notebookDir);
+      }
+
+      // some test profile does not build zeppelin-web.
+      // to prevent zeppelin starting up fail, create zeppelin-web/dist directory
+      new File("../zeppelin-web/dist").mkdirs();
+
+      LOG.info("Staring test Workbench server up...");
+
+      executor = Executors.newSingleThreadExecutor();
+      executor.submit(SERVER);
+      long s = System.currentTimeMillis();
+      boolean started = false;
+      while (System.currentTimeMillis() - s < 1000 * 60 * 3) {  // 3 minutes
+        Thread.sleep(2000);
+        started = checkIfServerIsRunning();
+        if (started == true) {
+          break;
+        }
+      }
+      if (started == false) {
+        throw new RuntimeException("Can not start Zeppelin server");
+      }
+      //ZeppelinServer.notebook.setParagraphJobListener(NotebookServer.getInstance());
+      LOG.info("Test Zeppelin stared.");
+    }
+  }
+
+  public static void startUpWithKnoxEnable(String testClassName) throws Exception {
+    start(true, testClassName, true, true);
+  }
+
+  public static void startUpWithAuthenticationEnable(String testClassName) throws Exception {
+    start(true, testClassName, false, true);
+  }
+
+  public static void startUp(String testClassName) throws Exception {
+    start(false, testClassName, false, true);
+  }
+
+  public static void startUp(String testClassName, boolean cleanData) throws Exception {
+    start(false, testClassName, false, cleanData);
+  }
+
+  private static String getHostname() {
+    try {
+      return InetAddress.getLocalHost().getHostName();
+    } catch (UnknownHostException e) {
+      LOG.error("Exception in WebDriverManager while getWebDriver ", e);
+      return "localhost";
+    }
+  }
+
+  public static void shutDown() throws Exception {
+    shutDown(true);
+  }
+
+  protected static void shutDown(final boolean deleteConfDir) throws Exception {
+
+    if (!WAS_RUNNING) {
+
+      LOG.info("Terminating test Zeppelin...");
+      WorkbenchServer.jettyWebServer.stop();
+      executor.shutdown();
+
+      long s = System.currentTimeMillis();
+      boolean started = true;
+      while (System.currentTimeMillis() - s < 1000 * 60 * 3) {  // 3 minutes
+        Thread.sleep(2000);
+        started = checkIfServerIsRunning();
+        if (started == false) {
+          break;
+        }
+      }
+      if (started == true) {
+        throw new RuntimeException("Can not stop Submarine workbench server");
+      }
+
+      LOG.info("Test Submarine workbench server terminated.");
+
+      if (deleteConfDir) {
+        // don't delete interpreter.json when recovery is enabled. otherwise the interpreter setting
+        // id will change after zeppelin restart, then we can not recover interpreter process
```

Comment:
Need change to `// id will change after workbench restart, then we can not recover interpreter process`?